src/transformers/generation_utils.py

```diff
@@ -895,9 +1241,30 @@ def sample(
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
             # forward pass to get next token
-            outputs = self(**model_inputs, return_dict=True)
+            outputs = self(
+                **model_inputs,
+                return_dict=True,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
             next_token_logits = outputs.logits[:, -1, :]
 
+            # Store logits, attentions and hidden_states when required
+            if return_dict_in_generate:
+                if output_scores:
+                    logits += (next_token_logits,)
```

Comment:
let's add the `scores` tensors here before they go into the `F.softmax` -> those are the important values usually